static/js/settings_org.js

```diff
@@ -394,106 +418,159 @@ function _set_up() {
-                    // We expect message_content_edit_limit_seconds was sent in the
-                    // response as well
-                    var data_message_content_edit_limit_minutes =
-                        Math.ceil(response_data.message_content_edit_limit_seconds / 60);
-                    if (response_data.allow_message_editing) {
-                        if (response_data.message_content_edit_limit_seconds > 0) {
-                            ui_report.success(
-                                i18n.t("Users can now edit topics for all their messages, and the content of messages which are less than __num_minutes__ minutes old.",
-                                       {num_minutes : data_message_content_edit_limit_minutes}),
-                                message_editing_status);
-                        } else {
-                            ui_report.success(i18n.t("Users can now edit the content and topics of all their past messages!"), message_editing_status);
-                        }
-                    } else {
-                        ui_report.success(i18n.t("Users can no longer edit their past messages!"), message_editing_status);
-                    }
-                    // message_content_edit_limit_seconds could have been changed earlier
-                    // in this function, so update the field just in case
-                    $("#id_realm_message_content_edit_limit_minutes").val(data_message_content_edit_limit_minutes);
+        var subsection = $(this).closest('.org-subsection-parent');
+        var properties_elements = get_subsection_property_elements(subsection);
+        var show_change_process_button = false;
+        _.each(properties_elements , function (elem) {
+            if (check_property_changed(elem)) {
+                show_change_process_button = true;
+            }
+        });
+
+        var change_process_button = subsection.find('.subsection-header .button');
+        change_process_button.first().text(i18n.t("Save"));
+        if (show_change_process_button) {
+            change_process_button.removeClass('hide').addClass('show');
+        } else {
+            change_process_button.removeClass('show').addClass('hide');
+        }
+    });
+
+    $('.organization').on('click', '.subsection-header .subsection-changes-discard button', function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        var properties_elements = get_subsection_property_elements(this);
+        _.each(properties_elements, function (element) {
+            element = $(element);
+            var property_name = element.attr('id').replace("id_", "").split('-').join('_');
+            if (typeof page_params[property_name] === 'boolean') {
+                // Trigger a click event because of dependent sub-settings
+                if (element.prop('checked') !== page_params[property_name]) {
+                    element.click();
```

Comment:
I think for flipping a checkbox back to its default state, it'd be better to use something lower-level that triggering a click event (e.g. just setting the prop).

Reply:
Yeah, I've first used the `prop` but for cases like `Users can edit their messages` we have to also toggle dependent field like `Message edit limit in minutes`. So, in that case `click()` event holds better. But yes, for just one case it's not good to trigger `click()` event on all other fields. Any better suggestion.

Reply:
And I think there is no better way to determine if any field is dependent on another currently.
@timabbott so should leave as it is?

Reply:
I see.  If that's the reasoning, just leave a comment like this;
```
We trigger a click event, rather than just changing the prop here, because that automatically handles the logic around enabling/disabling dependent sub-settings.
```